### PR TITLE
Pause for column family

### PR DESCRIPTION
**Pull request by irieksts**

### CODE REVIEW

**Comment by boneill42** [comment]
Rock on. Reviewing now.

---

Brian O'Neill
Lead Architect, Software Development
Health Market Science | 2700 Horizon Drive | King of Prussia, PA 19406
p: 215.588.6024blog: http://weblogs.java.net/blog/boneill42/
blog: http://brianoneill.blogspot.com/

On 2/9/12 3:22 PM, "Isaac Rieksts"
<reply+i-3162957-bdad5328a209990f48d97006a977517e64a2d65f-897229@reply.gith
ub.com> wrote:

**Comment by boneill42** [comment]
Hey...

This looks like you can pause a trigger, but when I re-enable the trigger
will it process the transactions it missed while paused?

I was thinking more along the lines of pausing triggers for a whole column
family.  Then, the log entries would remain until processing for that
column family was re-enabled at which time all the triggers would "catch
back up" with the transactions they missed.

-brian

---

Brian O'Neill
Lead Architect, Software Development
Health Market Science | 2700 Horizon Drive | King of Prussia, PA 19406
p: 215.588.6024blog: http://weblogs.java.net/blog/boneill42/
blog: http://brianoneill.blogspot.com/

On 2/9/12 3:22 PM, "Isaac Rieksts"
<reply+i-3162957-bdad5328a209990f48d97006a977517e64a2d65f-897229@reply.gith
ub.com> wrote:

**Comment by irieksts** [comment]
Yes, it leaves the messages on the commit log until the trigger is re-enabled.

**Comment by boneill42** [comment]
perfect.

On Thu, Feb 9, 2012 at 4:48 PM, Isaac Rieksts <
reply@reply.github.com

> wrote:
> 
> Yes, it leaves the messages on the commit log until the trigger is
> re-enabled.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/hmsonline/cassandra-triggers/pull/2#issuecomment-3896377

## 

Brian ONeill
Lead Architect, Health Market Science (http://healthmarketscience.com)
mobile:215.588.6024
blog: http://weblogs.java.net/blog/boneill42/
blog: http://brianoneill.blogspot.com/

File: src/main/java/com/hmsonline/cassandra/triggers/TriggerStore.java [hunk 1]
```diff
@@ -54,14 +60,25 @@ public static Trigger getTrigger(String triggerClass) throws Exception {
         ColumnParent parent = new ColumnParent(COLUMN_FAMILY);
         List<KeySlice> rows = getConnection(KEYSPACE).get_range_slices(parent, predicate, keyRange,
                 ConsistencyLevel.ALL);
+        rowsLoop:
```
Comment (boneill42) [inline src/main/java/com/hmsonline/cassandra/triggers/TriggerStore.java]: No continues allowed. Fix.

Comment (irieksts) [inline src/main/java/com/hmsonline/cassandra/triggers/TriggerStore.java]: There is nothing wrong with continues.

http://programmers.stackexchange.com/questions/58237/are-break-and-continue-bad-programming-practices

Labels are more debatable.  Using a label is a sign that a method is growing too big.  

Will remove the label.  Will not remove the continue.

**Pull request merged**